torch/_dynamo/guards.py

```diff
@@ -487,6 +487,20 @@ def FUNCTION_MATCH(self, guard: Guard):
         if guard.is_local():
             return self.ID_MATCH(guard)
 
+    def CLOSURE_MATCH(self, guard: Guard):
```

Comment:
I don't get how this is sound, code id is not enough?

Reply:
It's safe because of the reasoning in here https://github.com/pytorch/pytorch/pull/109427#issuecomment-1732538371